metricflow-semantics/metricflow_semantics/experimental/dsi/model_object_lookup.py

```diff
@@ -62,6 +63,27 @@ def time_dimension_name_to_grain(self) -> Mapping[str, TimeGranularity]:  # noqa
     def entity_lookup(self) -> EntityLookup:  # noqa: D102
         return EntityLookup(self._semantic_model.entities)
 
+    @cached_property
+    def dimension_lookup(self) -> DimensionLookup:  # noqa: D102
+        return DimensionLookup(self._semantic_model.dimensions)
+
+    @cached_property
+    def resolved_primary_entity_name(self) -> str:
```

Comment:
Our [primary entity validation](https://github.com/dbt-labs/dbt-semantic-interfaces/blob/main/dbt_semantic_interfaces/validations/primary_entity.py#L29) seems to only guarantee that semantic models with dimensions have a primary entity (i.e., if you have a model with entities only it doesn't need to have a primary entity). Can we update this code to avoid erroring if we encounter a model like that?

Reply:
We might not have any tests for this type of model, might be good to add one!